scheduler/repositories/sqlalchemy/task_store.py

```diff
@@ -19,17 +20,38 @@ def __init__(self, datastore: SQLAlchemy) -> None:
         self.datastore = datastore
 
     def get_tasks(
-        self, scheduler_id: Union[str, None], status: Union[str, None], offset: int = 0, limit: int = 100
+        self,
+        scheduler_id: Union[str, None],
+        type: Union[str, None],
+        status: Union[str, None],
+        offset: int = 0,
+        limit: int = 100,
+        min_created_at: Optional[datetime.datetime] = None,
+        max_created_at: Optional[datetime.datetime] = None,
+        filters: List[models.Filter] = None,
```

Comment:
The type hint or default value is misleading. If `filters` is truly nullable, then I would suggest making the type `Optional[List[Filter]]`. Another way could be using a default factory value (to create an empty list using `Field(default_factory=list`, ...)`. Personally, I would go for the latter since this has better semantic meaning than a nullable option + allowing an empty list.

Reply:
I agree using a default here is misleading.